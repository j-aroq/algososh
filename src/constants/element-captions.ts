export const HEAD = "head";
export const TAIL = "tail";
export const DIV_CIRCLES = 'div[class^="circle_circle"';
export const DIV_CIRCLE_HEAD = 'div[class*="circle_head"';
export const DIV_CIRCLE_TAIL = 'div[class*="circle_tail"';
export const DIV_CIRCLE_SMALL = 'div[class*="circle_small"';
export const INPUT_FIRST = "input:first";
export const INPUT_LAST = "input:last";
