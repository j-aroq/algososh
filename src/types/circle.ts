import { ElementStates } from "./element-states";

export type TCircle = {
  value?: string | null;
  state?: ElementStates;
  head?: string | null;
  tail?: string | null;
  extra_circle?:
    | {
        insertion?: boolean;
        removal?: boolean;
        value?: string | null;
        state?: ElementStates;
      }
    | undefined;
};
