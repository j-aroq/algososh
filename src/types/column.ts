import { ElementStates } from "./element-states";

export type TColumn = {
    number: number;
    state: ElementStates;
  }