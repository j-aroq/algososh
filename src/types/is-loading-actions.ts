export type TIsLoadingInsertion = {
  isLoading?: boolean;
  isLoadingInsertHead?: boolean;
  isLoadingInsertTail?: boolean;
  isLoadingInsertByIndex?: boolean;
};

export type TIsLoadingRemoval = {
  isLoading?: boolean;
  isLoadingRemoveHead?: boolean;
  isLoadingRemoveTail?: boolean;
  isLoadingRemoveFromIndex?: boolean;
};
