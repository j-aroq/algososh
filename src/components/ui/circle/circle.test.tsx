import renderer from "react-test-renderer";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";
import { HEAD, TAIL } from "../../../constants/element-captions";

const circleLetter = "A";

describe("Circle component", () => {
  it("Should render circle without letter", () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render circle with letter", () => {
    const tree = renderer.create(<Circle letter={circleLetter} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render circle with head", () => {
    const tree = renderer
      .create(<Circle letter={circleLetter} head={HEAD} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render circle with head react element", () => {
    const tree = renderer
      .create(<Circle letter={circleLetter} head={<Circle />} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render circle with tail", () => {
    const tree = renderer
      .create(<Circle letter={circleLetter} tail={TAIL} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render circle with tail react element", () => {
    const tree = renderer
      .create(<Circle letter={circleLetter} tail={<Circle />} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render circle with index", () => {
    const tree = renderer
      .create(<Circle letter={circleLetter} index={0} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render small circle", () => {
    const tree = renderer
      .create(<Circle letter={circleLetter} isSmall />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render circle with default state", () => {
    const tree = renderer
      .create(<Circle letter={circleLetter} state={ElementStates.Default} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render circle with changing state", () => {
    const tree = renderer
      .create(<Circle letter={circleLetter} state={ElementStates.Changing} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render circle with modified state", () => {
    const tree = renderer
      .create(<Circle letter={circleLetter} state={ElementStates.Modified} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
