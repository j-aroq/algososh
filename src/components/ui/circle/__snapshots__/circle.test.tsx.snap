// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component Should render circle with changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component Should render circle with default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component Should render circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component Should render circle with head react element 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component Should render circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle component Should render circle with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component Should render circle with modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component Should render circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Circle component Should render circle with tail react element 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Circle component Should render circle without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component Should render small circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
